docs/concepts/pe_scheme.rst

```diff
@@ -0,0 +1,187 @@
+Phase encoding scheme handling
+==============================
+
+From version 3.0RC1 onwards, *MRtrix3* is capable of importing information from
+DICOM relating to the phase encoding of the acquired images, and encoding this
+information within key-value fields inside an image header. This information can
+then later be used by the ``dwipreproc`` script, specifically using its
+``-rpe_header`` command-line option, to automatically set up and execute FSL's
+``topup`` and ``eddy`` commands without requiring explicit input from the user
+regarding the phase encoding design of the imaging experiment. This page
+explains how this information is encoded and manipulated by *MRtrix3*.
+
+
+.. NOTE::
+
+    Due to variations in sequences, acquisition protocols, DICOM encoding, and
+    image manipulation operations, *MRtrix3* cannot be guaranteed to obtain or
+    retain the phase encoding information correctly in all circumstances. Manual
+    inspection of these data and/or the outcomes of image processing when making
+    use of these data is advocated. If circumstances leading to incorrect import
+    or encoding of this information in *MRtrix3* is reproducible, please report
+    this to the *MRtrix3* developers so that the software can be updated to
+    support such data.
+
+
+Phase encoding information storage
+----------------------------------
+
+The phase encoding information for a particular image file can be stored in one
+of two ways.
+
+-  The most convenient of these is storage of (one or more) key-value field(s)
+   encapsulated within the :ref:`image_header`, just as can be used for
+   :ref:`diffusion_gradient_scheme_handling`. This ensures that the information
+   is retained through image processing, as each *MRtrix3* command passes the
+   header entries of the input image through to the output image.
+
+-  Alternatively, this information can be stored within a JSON file that
+   accompanies the relevant image file(s). This information would then typically
+   be imported / exported using the ``-json_import`` and ``-json_export`` options
+   in ``mrconvert``.
+
+Precisely *how* this phase encoding information is *encoded* however depends on
+the nature of the phase encoding information for that image; specifically,
+whether the phase encoding information is *identical* for all volumes within an
+image file (or if it contains just one volume), or whether the phase encoding
+information *varies* between volumes within the image series.
+
+
+Fixed phase encoding
+....................
+
+In the case where both the phase encoding *direction* and the EPI *total readout
+time* are equivalent for all volumes within an image, this information is encoded
+within two fields: "``PhaseEncodingDirection``" and "``TotalReadoutTime``". These
+fields are consistent with `BIDS
+<http://bids.neuroimaging.io//>`_ (the Brain Imaging Data Structure).
+
+``PhaseEncodingDrection`` can take one of six values: "``i``", "``i-``", "``j``",
```

Comment:
I find this section on the coordinate system implied a bit confusing... In NIfTI, the coordinate system can be angulated in such a way that the image axes no longer relate in any sense to anatomical RAS. What happens if the images are acquired e.g. in the coronal plane, with the PE direction inferior-superior? Does this still qualify as along `j` - which would be the column axis for the raw image slices, but clearly not A>>P...? I'm not familiar with this part of the code, so maybe I'm missing something...? 

Similarly, I also find the following note troublesome, given how MRtrix manipulates strides internally. For instance, a sagittal acquisition with PE A>>P would end up with the PE direction along rows of the raw images slices (so presumably `i`), which would be mapped onto `y` by the MRtrix strides/transform reshuffle (although with stride 1). So I find it difficult to reconcile such a scenario with the statement that images strides have no effect here... What happens to the PE information in such a case? Are the PE labels interpreted relative to the axes taking their strides into account - a bit like the NIfTI transform handling...?

Reply:
My wording here is probably not great. It's a difficult thing to describe at all, let alone for an audience less versed with such things, and maybe I tried to simplify it too far. Then again maybe my handling isn't right either...

`i`, `j` and `k` correspond strictly to the order of the image axes as they are stored within the NIfTI file.  Whenever the axes are re-jigged by *MRtrix3* to approximate axial, it also has to alter the indicated phase encoding direction (and slice encoding direction since #1169). You can see this happening in `Header::realign_transform()`. Indeed in the case of NIfTI, since such a transformation occurs purely within the image write function (`File::NIfTI*::write()` calling `File::NIfTI::adjust_transform()`) and doesn't alter the image header in any way, `File::JSON::save()` has to *predict* how the associated image will be / will have been transformed, and perform the corresponding transformation to the phase / slice encoding indicator(s); as does `File::JSON::load()` since such data are loaded into the header after the transformation has occurred.

I did acquire a set of phantom images with axial / coronal / sagittal slices and all 4 possible phase encoding directions for each, and in testing it all appeared to behave as it should. But it's entirely plausible that I got something wrong. e.g. I just realised that it's currently ignoring the `Header::do_not_realign_transform()` flag, though that only affects `mrinfo`. I'll see if I can fix up the wording here without confusing both myself and the reader. Feel free to modify yourself.

Reply:
Looking at the code, I think that `File::JSON::load()` should be duplicating the internals of `Header::adjust_transform()` rather than using `File::NIfTI::adjust_transform()` (if they may not have equivalent outcomes).

There's also #847, which I haven't fully convinced myself of. My suspicion is that people are going to manually decide what to add here based on `mrview` / `mrinfo`, which are going to report the image axes *after* *MRtrix3* has permuted them to near-axial, and need to correspond to a NIfTI that has been manually saved with `-stride -1,+2,+3,+4`; which I suppose means that L-R might need to be explicitly flipped in `dwipreproc`, and it probably needs further documentation on such... :hurtrealbad:

Reply:
@jdtournier See if ac67e07 is any better?